ci-metrics/ci_listener.py

```diff
@@ -272,6 +276,79 @@ def handle_tests(self, key, value, retried=False):
+                    self.message_out["%s_total_tests_exec" %
+                        executor] = int(tester["executed"]) + \
+                        self.message_out["%s_total_tests_exec" % executor]
+                else:
+                    self.message_out["%s_total_tests_exec" %
+                        executor] = int(tester["executed"])
+                if "total_tests_exec" in self.message_out.keys():
+                    self.message_out["total_tests_exec"] = \
+                        int(tester["executed"]) + \
+                        self.message_out["total_tests_exec"]
+                else:
+                    self.message_out["total_tests_exec"] = \
+                        int(tester["executed"])
+                if "%s_total_tests_failed" % executor \
+                    in self.message_out.keys():
+                        self.message_out["%s_total_tests_failed" %
+                            executor] = int(tester["failed"]) + \
+                            self.message_out["%s_total_tests_failed" % executor]
+                else:
+                    self.message_out["%s_total_tests_failed" %
+                        executor] = int(tester["failed"])
+                if "total_tests_failed" in self.message_out.keys():
+                    self.message_out["total_tests_failed"] = \
+                        int(tester["failed"]) + \
+                        self.message_out["total_tests_failed"]
+                else:
+                    self.message_out["total_tests_failed"] = \
+                        int(tester["failed"])
+                if "total_time_secs" in self.message_out.keys():
+                    self.message_out["total_time_secs"] = \
+                        max(int(time_end - time_start), 0) + \
+                        self.message_out["total_time_secs"]
+                else:
+                    self.message_out["total_time_secs"] = \
+                        max(int(time_end - time_start), 0)
+                if "total_time_mins" in self.message_out.keys():
+                    self.message_out["total_time_mins"] = \
+                        max(int(time_end - time_start), 0)/60.0 + \
+                        self.message_out["total_time_mins"]
+                else:
+                    self.message_out["total_time_mins"] = \
+                        max(int(time_end - time_start), 0)/60.0
+                if "total_time_hrs" in self.message_out.keys():
+                    self.message_out["total_time_hrs"] = \
+                        max(int(time_end - time_start), 0)/3600.0 + \
+                        self.message_out["total_time_hrs"]
+                else:
+                    self.message_out["total_time_hrs"] = \
+                        max(int(time_end - time_start), 0)/3600.0
+        return True
+
+    def handle_recipients(self, key, value, retried=False):
+        recipient_set = Set([])
+        # Add old recipients to the set
+        if "recipients" in self.message_out.keys():
```

Comment:
I would replace all of this with:

recipient_set = Set(map(lambda x: x.strip(), message_out.get("recipients").split(",")))